### PR TITLE
Allow specifying default capacities on actors themselves

### LINKED ISSUE #70: Let `Actor`s specify their default `Addr` capacities
Currently, if a user wants to create an actor with non-default (normal and/or high-priority) inbox capacities, they need to either create their `addr` first using `Addr::with_capacity()` or spawn them using `system.prepare[_fn]().with_capacity()`.

However, the actors themselves often have enough information to determine their best (or at least good default) inbox capacities, so let's allow them doing so! This has a potential to noticeably simplify "actor system setup" codebase sections.

`Addr` already has actor generic type parameter, so it can access static methods and constants defined on it in its constructor.

Implementation-wise this would be about adding either an associated constant to `Actor` trait to determine default capacity (if Rust allows having defaults for associated constants), or a static method (without `&self`) to the `Actor` trait with default implementation. In both cases we likely don't want to add any more _mandatory_ items that Actors need to implement. Change `impl Default for Addr` to use these method(s)/constant(s).

Another thing left to bikeshed is whether this should be a single constant/method returning `Capacity`, or 2 separate methods/constants for normal, high-prio capacities returning `usize`. Notice the `DEFAULT_CHANNEL_CAPACITY` constant is not currently public.

CC @bschwind @skywhale @PabloMansanet.

### PR DESCRIPTION
**Pull request by strohel**
I'm myself not convinced about this, but publishing for feedback anyway.

Resolves #70 (I apparently forgot about this PR when writing the issue; it even shows that defaults for associated constants are possible/stable).

### CODE REVIEW

**Comment by bschwind** [comment]
@strohel I've rebased this branch on the latest `main`

**Comment by bschwind** [comment]
Related to #70

**Comment by strohel** [comment]
Thanks, @bschwind - yeah we should get this done. Looks like you're on it?

**Comment by strohel** [comment]
@bschwind I think we can just merge this and cut a new version, WDYT?

File: src/lib.rs [hunk 1]
```diff
@@ -341,6 +341,8 @@ impl<'a, A: 'static + Actor<Context = Context<<A as Actor>::Message>>, F: FnOnce
     SpawnBuilder<'a, A, F>
 {
     /// Specify a capacity for the actor's receiving channel. Accepts [`Capacity`] or [`usize`].
+    ///
+    /// Ignored when `.with_addr()` is used at the same time.
```
Comment (skywhale) [inline src/lib.rs:345]: Thank you for adding this documentation :)

**Review by skywhale — APPROVED** [review]
LGTM

The changes make a lot of sense to me :)

**Review by bschwind — APPROVED** [review]
LGTM!

I don't believe this is a breaking change, so we can just do a patch release, right?

**Comment by strohel** [comment]
> I don't believe this is a breaking change, so we can just do a patch release, right?

I think/hope so, so let me jut cut it.

**Pull request merged**